### PR TITLE
Fix a small bug in Reader destructor

### PR DESCRIPTION
**Pull request by Kh4L**
Signed-off-by: Serge Panev <spanev@nvidia.com>

### CODE REVIEW

File: dali/pipeline/operators/reader/reader_op.h [hunk 1]
```diff
@@ -57,10 +57,12 @@ class DataReader : public Operator<Backend> {
...
   virtual ~DataReader() noexcept {
     StopPrefetchThread();
-    for (int i = 0; i < Operator<Backend>::batch_size_; ++i) {
-      // return unconsumed batches
-      if (prefetched_batch_[i]) {
-        loader_->ReturnTensor(prefetched_batch_[i]);
+    if (prefetched_batch_.size() > 0) {
+      for (int i = 0; i < Operator<Backend>::batch_size_; ++i) {
+        // return unconsumed batches
+        if (prefetched_batch_[i]) {
```
Comment (JanuszL) [inline dali/pipeline/operators/reader/reader_op.h]:
```
for (size_t i = 0; i < prefetched_batch_.size() ; ++i)
```

Comment (Kh4L) [reply dali/pipeline/operators/reader/reader_op.h]: Done

**Comment by Kh4L** [comment]
Build 40883

File: dali/pipeline/operators/reader/reader_op.h [hunk 2]
```diff
@@ -57,10 +57,12 @@ class DataReader : public Operator<Backend> {
 
   virtual ~DataReader() noexcept {
     StopPrefetchThread();
-    for (int i = 0; i < Operator<Backend>::batch_size_; ++i) {
-      // return unconsumed batches
-      if (prefetched_batch_[i]) {
-        loader_->ReturnTensor(prefetched_batch_[i]);
+    if (prefetched_batch_.size() > 0) {
```
Comment (JanuszL) [inline dali/pipeline/operators/reader/reader_op.h]: This is not needed now.

**Pull request merged**